cobra_commander/lib/cobra_commander/registry.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module CobraCommander
+  module Registry
+    def [](key)
+      Class.new(self) do
+        define_singleton_method(:key) { key }
+        define_method(:key) { key }
+
+        def self.inherited(base)
+          super
+          superclass.all[key] = base
+        end
+      end
+    end
```

Comment:
I'm going to have to open a console in here to unpack what this accomplishes. I wasn't familiar with [Forwardable](https://ruby-doc.org/stdlib-2.5.1/libdoc/forwardable/rdoc/Forwardable.html).

What is `key`? Is it a name or a path, or what?

Reply:
Yeah, I'm not a big fan of how this is right now, but what it achieves is:

I want a class extending the registry to be **registered in the registry** and **have a key attribute** used to identify the registered class uniquely within the registry. This key is used to toggle the plugin on/off in the, for instance.

The key is a required class attribute of the source, so I didn't want to just have them define a key, I wanted to initialize it with the inheritance. This is the best I could do.

This defines a class with a key, and once inherited it will register the new class with the given key. The resulting syntax is similar to what AR does with migrations. I wish the code was simpler, but I like the overall solution.

Reply:
Is it a `category` then perhaps? Or possibly a `klass`? --- Maybe even just a `registration_key`?